Feature: Add Typescript type checking of replies based on the status code. (#3097)
fastify/fastify: docs/Reply.md

If you want to know more please review [Routes#async-await](Routes.md#async-await).

<a name="send-reply-typed"></a>

> ### .sendReplyTyped(code, data)

Reviewer: I would prefer a more explicit `sendWithStatus(code, data)`

Author: The latest commit has renamed the method.

> As the name suggests, `.sendWithStatus()` is a function that sends a reply to the user along with a status code. When used in combination

Reviewer 2: ```suggestion
Sends a reply with a specific status code. When used in combination
```

Author: Addressed in latest commit.
